steampipecloud/table_steampipecloud_workspace_snapshot.go

```diff
@@ -0,0 +1,595 @@
+			},
+			{
+				Name:        "identity_handle",
+				Description: "The handle of the identity.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getIdentityWorkspaceDetails,
+			},
+			{
+				Name:        "identity_type",
+				Description: "The type of identity, can be org/user.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getIdentityWorkspaceDetails,
+			},
+			{
+				Name:        "workspace_id",
+				Description: "The id of the workspace.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "workspace_handle",
+				Description: "The handle of the workspace.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getIdentityWorkspaceDetails,
+			},
+			{
+				Name:        "state",
+				Description: "The current state of the snapshot.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "visibility",
+				Description: "The visibility of the snapshot.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "dashboard_name",
+				Description: "The mod-prefixed name of the dashboard this snapshot belongs to.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "schema_version",
+				Description: "The schema version of the underlying snapshot.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "inputs",
+				Description: "The inputs used for this snapshot.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "variables",
+				Description: "The variables used for this snapshot.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "search_path",
+				Description: "The search path used for this snapshot.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "start_time",
+				Description: "The time the dashboard execution started.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "end_time",
+				Description: "The time the dashboard execution ended.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "data",
+				Description: "The data for snapshot.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getSnapshotData,
+			},
+			{
+				Name:        "query_where",
+				Description: "The query where expression to filter snapshots.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromQual("query_where"),
+			},
+			{
+				Name:        "created_at",
+				Description: "The time when the snapshot was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "created_by_id",
+				Description: "ID of the user who created the snapshot.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "created_by",
+				Description: "Information about the user who created the snapshot.",
+				Type:        proto.ColumnType_JSON,
+			},
```

Comment:
Getting null value for `created_by_id`.
```suggestion
			{
				Name:        "created_by_id",
				Description: "ID of the user who created the snapshot.",
				Type:        proto.ColumnType_STRING,
				Transform:   transform.FromCamel(),
			},
			{
				Name:        "created_by_handle",
				Description: "Handle of the user who created the snapshot.",
				Type:        proto.ColumnType_STRING,
				Transform:   transform.FromField("CreatedBy.Handle"),
			},
```
We need to use transform `transform.FromCamel()` to get`created_by_id` value and instead of column `created_by` I think it is better to have a column `created_by_handle`.
```

```

Reply:
Added transform function for `created_by_id` and `updated_by_id` fields.
Also, it was decided that we will be mimicking the API field for metadata information i.e. no extra transform for `created_at`, `created_by_id`, `created_by`, `updated_at`, `updated_by_id`, `updated_by`, `version_id` fields.
The data for `created_by` and `updated_by` is received in an expanded object format containing information about the user who performed the respective operation.